fuzz/fuzz_targets/fuzz_int_set.rs

```diff
@@ -0,0 +1,673 @@
+    }
+}
+
+/* ### Length ### */
+struct LengthOp();
+
+impl LengthOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        (Some(Box::new(Self())), data)
+    }
+}
+
+impl Operation for LengthOp {
+    fn operate(&self, input: Input, _: Input) {
+        assert_eq!(input.int_set.len(), input.btree_set.len());
+    }
+
+    fn size(&self, _: usize) -> usize {
+        1
+    }
+}
+
+/* ### Is Empty ### */
+
+struct IsEmptyOp();
+
+impl IsEmptyOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        (Some(Box::new(Self())), data)
+    }
+}
+
+impl Operation for IsEmptyOp {
+    fn operate(&self, input: Input, _: Input) {
+        assert_eq!(input.int_set.is_empty(), input.btree_set.is_empty());
+    }
+
+    fn size(&self, _: usize) -> usize {
+        1
+    }
+}
+
+/* ### Contains ### */
+struct ContainsOp(u32);
+
+impl ContainsOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        let (Some(val), data) = read_u32(data) else {
+            return (None, data);
+        };
+        (Some(Box::new(Self(val))), data)
+    }
+}
+
+impl Operation for ContainsOp {
+    fn operate(&self, input: Input, _: Input) {
+        assert_eq!(
+            input.int_set.contains(self.0),
+            input.btree_set.contains(&self.0)
+        );
+    }
+
+    fn size(&self, length: usize) -> usize {
+        length.ilog2() as usize
+    }
+}
+
+/* ### Clear  ### */
+
+struct ClearOp();
+
+impl ClearOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        (Some(Box::new(Self())), data)
+    }
+}
+
+impl Operation for ClearOp {
+    fn operate(&self, input: Input, _: Input) {
+        input.int_set.clear();
+        input.btree_set.clear();
+    }
+
+    fn size(&self, length: usize) -> usize {
+        length
+    }
+}
+
+/* ### Intersects Range ### */
+
+struct IntersectsRangeOp(u32, u32);
+
+impl IntersectsRangeOp {
+    fn parse_args(data: &[u8]) -> (Option<Box<dyn Operation>>, &[u8]) {
+        let (Some(min), data) = read_u32(data) else {
+            return (None, data);
+        };
+        let (Some(max), data) = read_u32(data) else {
+            return (None, data);
+        };
+        (Some(Box::new(Self(min, max))), data)
+    }
+}
+
+impl Operation for IntersectsRangeOp {
+    fn operate(&self, input: Input, _: Input) {
+        let int_set_intersects = input.int_set.intersects_range(self.0..=self.1);
+
+        let mut btree_intersects = false;
+        for v in self.0..=self.1 {
+            if input.btree_set.contains(&v) {
+                btree_intersects = true;
+                break;
+            }
+        }
+
+        assert_eq!(int_set_intersects, btree_intersects);
+    }
+
+    fn size(&self, length: usize) -> usize {
+        if self.1 < self.0 {
+            return 1;
+        }
+        ((self.1 as usize - self.0 as usize) + 1) * (length.ilog2() as usize)
+    }
+}
+
+/* ### First  ### */
+
+struct FirstOp();
```

Comment:
I wonder if these might not be more cleanly expressed as an enum? but probably not worth changing at this point.

Reply:
I originally did this as an enum, but there ends up being more boiler plate (eg. match statements) needed then versus using dyn + a trait.

Reply:
I'm reading this and thinking the same thing.